Objects/genobject.c

```diff
@@ -354,15 +354,37 @@ gen_close(PyGenObject *gen, PyObject *args)
     PyObject *yf = _PyGen_yf(gen);
     int err = 0;
 
+    if (gen->gi_frame_state == FRAME_CREATED) {
+        gen->gi_frame_state = FRAME_COMPLETED;
+        Py_RETURN_NONE;
```

Comment:
You _could_ drop this line and everything should work the same. Probably about the same speed too (the memory locations read on the next line will be in the L1 cache).

Reply:
I think it is more readable this way, so I'll leave it.